### ISSUE TITLE
[Bug]: Wrong internal links in documents preview when using site(s)

### ISSUE THREAD

**Issue by olivermerz** [pimcore/pimcore#15336]
### Pimcore version

10.5.23

### Steps to reproduce

When opening a document within a site in preview mode wrong (internal) links are parsed. The link starts with the path/key of the site. When clicking the link, a 404 error appears because the route does not exist.

Same behavior for links in a WYSIWYG editable or in a link editable.

Example:
```
Home
|
└───Website (defined as site with given domain)
    |
    └───Page 1
    │   │   Page 1.1
    │   │   Page 1.2
    │   │   ...
    │
    └───Page 2
        │   Page 2.1
        │   Page 2.2
        │   ...
```

A link to Page 1.2 is parsed in preview (both embedded or new window) as `/Website/Page 1/Page 1.2`.

When opening the page **not** in preview mode the link is parsed correctly as `/Page 1/Page 1.2`.

### Expected / actual behavior

When opening the page in preview, the link should also be parsed as `/Page 1/Page 1.2`, so that the link can also be clicked in preview mode without getting an error.

**Comment by mattamon** [comment]
@olivermerz 
Thank you for sharing this issue.
Could you please describe the step to reproduce a little more detailed. At the moment it is very abstract and trying to reproduce it could be a problem. 
You can always refer to the demo. Also screenshots would help. 

Thank you!

**Comment by kingjia90** [comment]
@mattamon if i got correctly, you can reproduce by: 
- converting `de` to sub-site, set main domain to eg. `de.pimcore.fun`
- edit a link (eg. in `home` below carousel), drag and drop a document under the sub-site `de`
![image](https://github.com/pimcore/pimcore/assets/6014195/d421b7ad-0ac3-44b5-8c27-c866c1bf9f7b)
- in preview the link is appearing as `https://demo.pimcore.fun/de/Magazin` (from a main site point of view in the tree), Oliver means that would expect the url to be `https://de.pimcore.fun/Magazin`, so that when you visit the website (not preview mode) it appears as relative `/Magazin` (starting from `/de` instead of `Home`)

**Comment by olivermerz** [comment]
> @mattamon if i got correctly, you can reproduce by:
> 
> * converting `de` to sub-site, set main domain to eg. `de.pimcore.fun`
> * edit a link (eg. in `home` below carousel), drag and drop a document under the sub-site `de`
>   ![image](https://user-images.githubusercontent.com/6014195/247863992-d421b7ad-0ac3-44b5-8c27-c866c1bf9f7b.png)
> * in preview the link is appearing as `https://demo.pimcore.fun/de/Magazin` (from a main site point of view in the tree), Oliver means that would expect the url to be `https://de.pimcore.fun/Magazin`, so that when you visit the website (not preview mode) it appears as relative `/Magazin` (starting from `/de` instead of `Home`)

@kingjia90 Exactly, and then when you click the link in the preview, you get a 404 error.

**Comment by kingjia90** [comment]
Tried to fix it but clicking on the link in there it seems that would throw Content Security Policy error 🤔 
https://github.com/pimcore/pimcore/compare/10.6...fix-multisite-link?expand=1
May need to allow the devs to disable/enable CSP before using this feature

**Comment by dvesh3** [comment]
Fixed by #15724
